Домашнее задание 9 спринт

Nataliya2288 opened the pull request:
Направляю на проверку

Nataliya2288 committed:
Домашнее задание 9 спринт

Charmandik commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,75 @@
+    }
+
+    private var inputText: String? = null
+    private lateinit var inputEditText: EditText
```
Вместо использования lateinit, я рекомендую тебе использовать подход с добавлением nullability к типу объекта с оператором ?. и сразу проинициализировать его значением null. Это решение позволяет избежать ошибок, связанных с попытками доступа к неинициализированным объектам, которые могут привести к непредсказуемому поведению или крашам приложения. Хотя это может добавить некоторую дополнительную работу при обращении к объекту через оператор ?. , но это значительно повысит безопасность приложения, так как ты будешь уверен, что объект всегда либо имеет значение, либо равен null, и не будет вызывать исключительных ситуаций.Можешь еще прочитать статью на эту тему https://habr.com/ru/companies/maxilect/articles/590061

Charmandik commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,75 @@
+            val imm: InputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            imm.hideSoftInputFromWindow(window.decorView.windowToken, 0)
+        }
+        val simpleTextWatcher = object : TextWatcher {
```
Можно использовать функцию расширения doOnTextChanged вместо класса TextWatcher для отслеживания изменений текста в EditText. Вот пример использования:

editText.doOnTextChanged { text, start, before, count ->
    // Здесь можно выполнять действия при изменении текста    
// text - новый текст
    // start - начальная позиция изменения    
// before - количество символов, удаленных из текста
    // count - количество добавленных символов в текст}

Метод doOnTextChanged принимает лямбда-выражение, которое будет вызываться при каждом изменении текста. Внутри этого выражения можно выполнять необходимые действия в соответствии с изменениями текста.

Charmandik commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,70 +4,38 @@
+        android:textColor="?attr/colorPrimary"
+        android:textSize="@dimen/text_normal" />
+    <Button
+        android:id="@+id/search"
```
Для более простого использования различных классов view рекомендую использовать следующий трюк: К имени переменной добавлять префикс в виде "аббревиатуры" класса, к которой она относится, для TextView - tv, ImageView - iv, RecyclerView - rv и т.д.. Например, у тебя есть RecyclerView для треков, в этом случае будет rvTracks

Charmandik commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,6 +1,79 @@
+        android:fontFamily="@font/ys_display_medium"
+        android:text="@string/search"
+        android:textColor="?attr/colorOnSecondary"
+        android:textFontWeight="500"
```
шрифт и цвет стоит перенести в стиль, здесь же только оставить параметры расположения элемента

Nataliya2288 committed:
Домашнее задание 9 спринт
Правки:
1. Исправила кнопку "написать в поддержку",на эмуляторе теперь открывается почтовый клиент
2. Убрала часть кода в грэдл, который ссылается на LIST
